src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -147,7 +147,8 @@ await functionMetadata
             // we need to sync triggers if config changed, or the files changed
             await _functionsSyncManager.TrySyncTriggersAsync();
 
-            (var success, var functionMetadataResult) = await TryGetFunction(name, request, configChanged);
+            // Setting force refresh to false as host restart causes a refersh already
```

Comment:
There's a spelling error here. Also, you mention a host restart - why are you guaranteed that actually happens? E.g. file change notifications may be disabled for the app, meaning file changes don't cause a restart.